xformers/factory/model_factory.py

```diff
@@ -277,7 +277,9 @@ def forward(
 
                 # Apply the optional input masking
                 if encoder_input_mask is not None:
-                    x += encoder_input_mask.unsqueeze(0).unsqueeze(-1)
+                    if x.dim() - encoder_input_mask.dim() > 1:
```

Comment:
I had to add this check to avoid a tensor shape mismatch error.

Reply:
oh, well done, thanks for fixing that